editor/src/core/model/project-file-utils.ts

```diff
@@ -499,7 +499,12 @@ export function canUpdateRevisionsState(
     case RevisionsState.ParsedAhead:
       return updated === RevisionsState.ParsedAhead || updated === RevisionsState.BothMatch
     case RevisionsState.CodeAhead:
-      return updated === RevisionsState.CodeAhead || updated === RevisionsState.BothMatch
+    case RevisionsState.CodeAheadButPleaseTellVSCodeAboutIt:
+      return (
+        updated === RevisionsState.CodeAhead ||
+        updated === RevisionsState.BothMatch ||
+        updated === RevisionsState.CodeAheadButPleaseTellVSCodeAboutIt
+      )
```

Comment:
I don't know if this matters, so I'm only pointing it out in case you then realise it does...
With this fall through in the switch statement, it means that the `RevisionsState` can be updated from `CodeAheadButPleaseTellVSCodeAboutIt` to `CodeAhead`.